### ISSUE TITLE
forwardedHeaders does not appear to work in 1.4.2

### ISSUE THREAD

**Issue by c-knowles** [containous/traefik#2364]
### What did you do?

I tried to setup a list trusted IPs in front of Traefik which are allowed to forward the `X-Forwarded-*` headers. The setup is a k8s Ingress with an IP whitelist in the Ingress. Adding trusted IPs as per the documentation does not seem to work, only marking it as insecure or whitelisting `0.0.0.0/0` in the Ingress allows traffic in.

### What did you expect to see?

`trustedIPs` inside `entryPoints.http.forwardedHeaders` should allow the IPs in the list to be trusted to send `X-Forwarded-*` HTTP headers.

### What did you see instead?

`trustedIPs` inside `entryPoints.http.forwardedHeaders` does not seem to have much effect at all.

### Output of `traefik version`: (_What version of Traefik are you using?_)

v1.4.2

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

I'm setting this up via the Helm chart in Kubernetes. Relevant part of the config is:

```toml
logLevel = "INFO"
defaultEntryPoints = ["http","https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
      entryPoint = "https"
    [entryPoints.http.forwardedHeaders]
      trustedIPs = ["10.2.0.0/16"]
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.forwardedHeaders]
      trustedIPs = ["10.2.0.0/16"]
[kubernetes]
```

`10.2.0.0/16` is the IP range of pods within my k8s cluster (kube-proxy based traffic would get routed from this range).

Note, the latest stable Helm chart does not support `forwardedHeaders` yet - I was working on that when I found this problem.

### Additional info.

The reason this appears to be Traefik side is I ran `brndnmtthws/nginx-echo-headers` from docker hub with a very basic k8s Ingress backed by Traefik and I get this back from it when the setup has `ingress.kubernetes.io/whitelist-source-range: "0.0.0.0/0"` regardless of what is in the Traefik toml config:

```
GET / HTTP/1.1
Host: echo-test.OBFUSCATED
User-Agent: curl/7.54.0
Accept: */*
X-Forwarded-For: MY_IP_OBFUSCATED, 10.2.12.0
X-Forwarded-Host: echo-test.OBFUSCATED
X-Forwarded-Port: 443
X-Forwarded-Proto: https
X-Forwarded-Server: traefik-master-traefik-2076230234-bxss6
X-Real-Ip: 10.2.12.0
Accept-Encoding: gzip
```

It shows `MY_IP_OBFUSCATED` with the correct external source IP address where I hit it, followed by an IP in the range I'm trying to whitelist - this IP varies depending on the request but always in the `10.2.0.0/16` range.

If I then set `ingress.kubernetes.io/whitelist-source-range: "MY_IP_OBFUSCATED/32"` in the Ingress I start to get HTTP 403 from Traefik.

Side note - I also do not see the 403s in the recent errors UI but I do see them in Traefik metrics. Probably a separate problem.

**Comment by ldez** [comment]
@c-knowles could you try this?

```toml
logLevel = "INFO"
defaultEntryPoints = ["http","https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
      entryPoint = "https"
    [entryPoints.http.forwardedHeaders]
      insecure = false
      trustedIPs = ["10.2.0.0/16"]
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.forwardedHeaders]
      insecure = false
      trustedIPs = ["10.2.0.0/16"]

[kubernetes]
```

**Comment by c-knowles** [comment]
@ldez I tried adding `insecure = false` to those two blocks but the behaviour appears to be the same unfortunately.

**Comment by c-knowles** [comment]
I started to look through the code in https://github.com/containous/traefik/pull/2262/files but could not immediately spot any issue.

**Comment by beenanner** [comment]
I just tested this with Google's LB and it appears that the X-Forwarded-For  header isn't pulling the first IP off the list for my use-case as the first IP is my pubic address and the last 2 are Google LBs

```
Hostname: 8a592c4309f1
IP: 127.0.0.1
IP: 10.0.7.32
IP: 10.0.7.14
IP: 172.18.0.5
GET / HTTP/1.1
Host: whoami.test.com
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9
Upgrade-Insecure-Requests: 1
Via: 1.1 google
X-Cloud-Trace-Context: f11cc2e3cce29b0f364b64061b330383/13854514456674479385
X-Forwarded-For: 170.20.11.15, 35.190.20.125, 130.211.3.126
X-Forwarded-Host: whoami.test.com
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: fc451219e95c
X-Real-Ip: 130.211.3.126
````

**Comment by beenanner** [comment]
This doesn't appear to be specific to k8 either as the above was setup on Docker Swarm Mode.

**Comment by emilevauge** [comment]
@c-knowles, is this one still an issue ?

**Comment by c-knowles** [comment]
@emilevauge I believe so, I've not managed to get this working properly yet so moved onto other problems. Has there been any significant changes in this part of Traefik? If so, I could give a newer version a go.

**Comment by ldez** [comment]
Closed by #2981
